target_snowflake/db_sync.py

```diff
@@ -309,6 +309,37 @@ def __init__(self, connection_config, stream_schema_message=None, table_cache=No
 
         self.snowpipe_on_error = self.connection_config.get('on_error')
 
+    def validate_stage_bucket(self, s3_bucket, stage):
+        """Validate that S3 bucket and external stage are correctly stated"""
+        self.logger.info(
+            f"Validating s3_bucket '{s3_bucket}' is stated correctly for the stage '{stage}'"
+        )
+        stage_name = stage.split('.')[1]
+        stage_query = f"SHOW STAGES LIKE '{stage_name}';"
```

Comment:
This place might be a reason of unexpected behaviour in future because stage names can be the same between different DB/Schema. For example we have `ADSTRA_STAGE` stage in `LOAD` and `DEV_LOAD` databases. 

Maybe we can get an s3 bucket in some other way? Or can we recognise stage more precisely?